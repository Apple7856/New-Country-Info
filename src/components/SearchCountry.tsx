import { Button, Container, TextField } from '@material-ui/core';
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

const SearchCountry = () => {
  const navigate = useNavigate();
  const [inputData, setInputData] = useState<string | null>(null);
  const getCountryInfo = () => {
    navigate("/country-info", { state: inputData });
  }
  return <Container>
      <form>
      <TextField data-testid="inputField" id="input" label="Enter Country Name" value={inputData} onChange={(e: any) => setInputData(e.target.value)} />
      <Button variant='contained' data-testid="button" color="primary" disabled={inputData ? false : true} onClick={()=>getCountryInfo()} >Search Country Info</Button>
      </form>
  </Container>
  ;
};

export default SearchCountry;
