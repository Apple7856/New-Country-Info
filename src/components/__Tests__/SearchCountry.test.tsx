import SearchCountry from '../SearchCountry';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

beforeEach(() => {
    document.body.innerHTML = ""
})

afterEach(() => {
    cleanup()
});

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => {
    const originalRouterDom = jest.requireActual('react-router-dom');
    return {
        __esModule: true,
        ...originalRouterDom,
        useNavigate: () => mockedNavigate
    };
});

describe("SearchCountry Component", () => {
    test("Input test", () => {
        const { getByTestId } = render(<SearchCountry />);
        const inputField = getByTestId('inputField');
        expect(inputField).toBeTruthy();
    })
    test("Button test", () => {
        const { getByTestId } = render(<SearchCountry />);
        const button = getByTestId('button');
        expect(button).toBeTruthy();
    })
    test("Snapshot test", () => {
        const component = renderer.create(<SearchCountry />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})