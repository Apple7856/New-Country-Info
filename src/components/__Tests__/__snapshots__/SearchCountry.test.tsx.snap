// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCountry Component Snapshot test 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <form>
    <div
      className="MuiFormControl-root MuiTextField-root"
      data-testid="inputField"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="input"
        id="input-label"
      >
        Enter Country Name
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value={null}
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      data-testid="button"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Search Country Info
      </span>
    </button>
  </form>
</div>
`;
